### PR TITLE
[improvement](jdbc catalog) Delete unnecessary schema and optimize insert logic

### PR DESCRIPTION
**Pull request by zy-kkk**
## Proposed changes

Issue Number: close #xxx

In the previous design, we were compatible with MySQL's auto-increment column and default value to bypass the null value check when writing back Jdbc External Table. However, because MySQL's default value is not completely unified with Doris, this resulted in The unsuitable default value is wrong. In response to this situation, I made the following optimizations
1. For JDBC External Table, we always allow certain columns to be missing during insertion. Even if these columns are not allowed to be empty at the source end, the error should be generated by the source end, not Doris herself.
2. When the target column is non-nullable and the insertion is done via `INSERT INTO tbl VALUES()` or `INSERT INTO tbl SELECT constants`, Doris should verify any inconsistency between them and throw an exception. This check is not applied for `INSERT INTO tbl SELECT ... FROM tbl` operations.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/NativeInsertStmt.java [hunk 1]
```diff
@@ -664,7 +664,9 @@ private void analyzeSubquery(Analyzer analyzer, boolean skipCheck) throws UserEx
         }
 
         // Check if all columns mentioned is enough
-        checkColumnCoverage(mentionedColumns, targetTable.getBaseSchema());
+        if (!(targetTable instanceof JdbcTable)) {
```
[the remark this answers is not in the document]
Comment (zy-kkk) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/NativeInsertStmt.java:668]: done

File: fe/fe-core/src/main/java/org/apache/doris/analysis/NativeInsertStmt.java [hunk 2]
```diff
@@ -698,6 +715,18 @@ private void analyzeSubquery(Analyzer analyzer, boolean skipCheck) throws UserEx
                         skipCheck);
                 // rows may be changed in analyzeRow(), so rebuild the result exprs
                 selectStmt.getResultExprs().clear();
+
+                if (targetTable instanceof JdbcTable) {
+                    for (int colIdx = 0; colIdx < targetColumns.size(); ++colIdx) {
+                        Column column = targetColumns.get(colIdx);
+                        Expr expr = rows.get(0).get(colIdx);
+                        if (!column.isAllowNull() && expr.isNullable()) {
```
[the remark this answers is not in the document]
Comment (zy-kkk) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/NativeInsertStmt.java]: For insert into tbl select constants, checking for null during the analysis phase is appropriate, but changing expr.isNullable() to expr instanceof NullLiteral makes it more reasonable

**Review by morningman — APPROVED** [review]
LGTM

**Review by wuwenchi — APPROVED** [review]

**Review by JNSimba — APPROVED** [review]

**Pull request merged**